### PR TITLE
build: migrate from travis-ci to github actions

### PR DESCRIPTION
**Pull request by lukehsiao**
This switches our CI/CD from Travis-CI to GitHub Actions.

For documentation purposes, some of the reasons we wanted to switch are:
* tighter integration with GitHub
* better performance (faster tests by a few mins)
* more resources for VMs (e.g., memory, cores, esp. for macOS)

There are some downsides in that the documentation is not as good as Travis-CI, a commit must be used to trigger a build, and there are some quirks with the environments (e.g., using the tmate action doesn't give you the same environment as just running GitHub action directly). But, overall, it seems like a reasonable switch.

### CODE REVIEW

**Comment by lukehsiao** [comment]
With this PR, using default snorkel, we have the following errors:

```
ERROR: tensorboard 2.2.0 has requirement setuptools>=41.0.0, but you'll have setuptools 40.6.2 which is incompatible.
ERROR: snorkel 0.9.3 has requirement torch<1.2.0,>=1.1.0, but you'll have torch 1.4.0+cpu which is incompatible.
```
Setuptools can be fixed just be updating the environment, I believe.

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies for Ubuntu
+      run: |
+        sudo apt-get install neofetch
+        sudo apt-get install -q -y libmecab-dev swig mecab
+        # In Ubuntu 18.04, unidic-mecab is too old, so we install it manually. Note
+        # that this automatically sets the dictionary to unidic.
```
Comment (HiromuHota) [inline .github/workflows/pythonpackage.yml:31]: This is actually not true.
The reason is that the unidic-mecab that will be installed on 18.04 by default does not include a dictionary as described one line below.

File: .github/workflows/pythonpackage.yml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+jobs:
+  ubuntu-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7]
+    services:
+      postgres:
+        image: postgres:12
```
[the remark this answers is not in the document]
Comment (lukehsiao) [reply .github/workflows/pythonpackage.yml:13]: Yeah, me too. I have postgres 12 locally which also seems to work fine.

[the remark this answers is not in the document]
Comment (HiromuHota) [reply .github/workflows/pythonpackage.yml:13]: If it works fine, I don't mind using pg12.

File: .github/workflows/pythonpackage.yml (hunk 1 above)
Comment (lukehsiao) [reply .github/workflows/pythonpackage.yml:31]: Thanks, I've updated the comment.

**Pull request merged**